pennylane/math/quantum.py

```diff
@@ -606,3 +606,166 @@ def _compute_mutual_info(
     )
 
     return vn_entropy_1 + vn_entropy_2 - vn_entropy_12
+
+
+def fidelity(state0, state1, check_state=False, c_dtype="complex128"):
+    r"""Compute the fidelity for two states (a state can be a state vector or a density matrix) acting on quantum
+    systems with the same size. For two pure states, the fidelity corresponds to the squared overlap. For a pure state
+    and a mixed state, it corresponds to the squared expectation of the mixed state in the pure state. Finally for two
+    mixed states, it is defined by the last formula:
+
+    .. math::
+        F( \ket{\psi} , \ket{\phi}) = \left|\bra{\psi}\ket{\phi}\right|^2
+
+        F( \ket{\psi} , \sigma ) = \left|\bra{\psi} \sigma \ket{\psi}\right|^2

```

Comment:
Should be `F( \ket{\psi} , \sigma ) = \bra{\psi} \sigma \ket{\psi}` (no square).

Reply:
Correct! 👍